File: project/frontend-services/libs/data-decoder/mapper/src/lib/mapper/DataDecoderMapper.ts
import {SensorTypeMapper} from './SensorTypeMapper';
import {DataDecoder, ScriptContent} from "@frontend-services/data-decoder/model";
import {DataDecoderDTO, DataDecoderResultDTO} from "@frontend-services/data-decoder/dto";

export class DataDecoderMapper {
  static dtoToModel(dto: DataDecoderResultDTO): DataDecoder {
    const sensorTypeId = SensorTypeMapper.dtoToModel(dto.data);
    const script = new ScriptContent(dto.script);
    const d = new Date(0); // The 0 there is the key, which sets the date to the epoch
    d.setUTCMilliseconds(dto.lastTimeSeen);
    return new DataDecoder(sensorTypeId, script, d);
  }

  static modelToDto(model: DataDecoder): DataDecoderDTO {
    return {data: {type: model.data.type}, script: model.script.content};
  }
}

File: project/backend-services/sharespot-data-decoder-master-backend/application/src/main/java/sharespot/services/data/decoder/master/application/DataDecoderMapper.java
package sharespot.services.data.decoder.master.application;

import sharespot.services.data.decoder.master.domain.DataDecoder;
import sharespot.services.data.decoder.master.domain.SensorTypeId;

public interface DataDecoderMapper {

    DataDecoder dtoToDomain(DataDecoderDTO dto);

    DataDecoderDTO domainToDto(DataDecoder domain);

    SensorTypeId dtoToDomain(SensorTypeIdDTO dto);

    SensorTypeIdDTO domainToDto(SensorTypeId domain);

}
